src/arrayformat.rs

```diff
@@ -11,9 +11,203 @@ use super::{
     Data,
     Dimension,
     NdProducer,
+    Ix
 };
 use crate::dimension::IntoDimension;
 
+#[derive(Debug)]
+enum ArrayDisplayMode {
+    // Array is small enough to me printed without omitting any values.
+    Full,
+    // Omit central values of the nth axis. Since we print that axis horizontally, ellipses
+    // on each row do something like a split of the array into 2 parts vertically.
+    VSplit,
+    // Omit central values of the certain axis. Since we do it only once, ellipses on each row
+    // do something like a split of the array into 2 parts horizontally.
+    HSplit(Ix),
+    // Both `VSplit` and `HSplit` hold.
+    DoubleSplit(Ix),
+}
+
+const PRINT_ELEMENTS_LIMIT: Ix = 5;
+
+impl ArrayDisplayMode {
+    fn from_array<A, S, D>(arr: &ArrayBase<S, D>, limit: usize) -> ArrayDisplayMode
```

Comment:
Looking at how you use it, I think you could take just the shape of the array as input, instead of the whole array.